platforms/core-configuration/model-core/src/main/java/org/gradle/api/internal/provider/AbstractCollectionProperty.java

```diff
@@ -121,6 +121,33 @@ public void addAll(Provider<? extends Iterable<? extends T>> provider) {
         getExplicitValue().addAll(provider);
     }
 
+    @Override
+    public void append(T element) {
```

Comment:
🤔 Perhaps it'd be reasonable to use `insert` / `insertAll` here as well?

Reply:
Maybe... I thought `insert` on lists usually involves a position, while `append` means add at the end, which is exactly what this method (and `add`) does. For maps/dictionaries, "`insert` this value for this key" seems reasonable. I could more easily agree that `add(key, value)` would be a decent alternative, but then there might be confusion with the  `add(value)` on lists/sets, which we ant to deprecate.